auth/README.md

```diff
@@ -0,0 +1,148 @@
+# Payment Gateway Authenticator
+
+This package provides a Key-based and JWT-based authentication system to The Graph Market and services
+
+## Features
+
+- **JWT Token Validation**: Validates JWT tokens using JWK sets fetched from a URL or provided as base64-encoded keys
+- **API Key Exchange**: Automatically exchanges API keys for JWT tokens via an issue endpoint
+- **Token Reissue**: Automatically reissues JWT tokens that are older than a configured threshold (too old to trust)
+- **Feature Configurations**: Extracts and propagates feature configurations from JWT claims
+- (No cutoff mechanism): Cutoff mechanism will be implemented in the upcoming 'Session' management plugin
+
+## Configuration
+
+The authenticator is configured using a URL-style connection string with the following format:
+
+```
+tgm://[host[:port]]?[parameters]
+```
+
+Default host: `auth.thegraph.market`
+
+### Parameters
+
+- `pubkeyurl`: URL to fetch the JWK set for JWT verification (default: `https://auth.thegraph.market/.well-known/jwks.json`)
+- `pubkeybase64`: Base64-encoded JWK set for offline JWT verification (mutually exclusive with `pubkeyurl`)
+- `reissue-jwt-max-age-secs`: Maximum age in seconds before a JWT token is reissued (default: 600)
+- `key`: Authentication key for the reissue endpoint to prevent rate limiting
+- `insecure`: Skip certificate verification (default: false)
+- `plaintext`: Use HTTP instead of HTTPS (default: false)
+
+### Example Configurations
+
+**Production:**
+```
+tgm://?key=server_key
+```
+
+**Development:**
+```
+tgm://localhost:8080?plaintext=true&pubkeyurl=http://localhost:8080/.well-known/jwks.json
+```
+
+## Usage
```

Comment:
We should show usage with `firecore` example first and another usage section in library mode (less frequent).